### PR TITLE
Create test suite

### PR DESCRIPTION
**Pull request by brandl-muc**
Add 97 UnitTests which need [jasmine-gjs](https://github.com/ptomato/jasmine-gjs) to execute.
Provides a script to install jasmine-gjs, but no makefile integration

### CODE REVIEW

**Comment by brandl-muc** [comment]
Another step to solve issue #25

File: test/install-jasmine-gjs.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+set -ex
+
+cd /tmp
```
Comment (da2x) [inline test/install-jasmine-gjs.sh]: Please use `mktemp -D` instead of a hardcoded path where we might run into conflicts.

**Comment by da2x** [comment]
This is looking great! Thanks for all the work you’ve put into this. You can assume jsamine-gjs is installed and add a “test” target to the makefile. Development dependencies can be added to an extra section underneath the runtime dependencies in the readme file.

File: test/install-jasmine-gjs.sh (hunk 1 above)
Comment (brandl-muc) [reply test/install-jasmine-gjs.sh]: Good idea, will do

Comment (brandl-muc) [reply test/install-jasmine-gjs.sh]: Done

**Comment by brandl-muc** [comment]
Sorry for the occasional progress, always so much else to do.
I should have integrated all your comments, please re-evaluate.

**Comment by brandl-muc** [comment]
@da2x Do you find some minutes to re-evaluate this? I don't want to continue before this is merged.

**Comment by da2x** [comment]
Merged. Thanks for all the tests and I’m sorry about the delay from my side.

**Pull request merged**